src/api/spec/controllers/kerberos_login_spec.rb

```diff
@@ -0,0 +1,98 @@
+require 'rails_helper'
+require 'gssapi'
+
+RSpec.describe 'Kerberos login', vcr: false, type: :request do
+  describe 'authentication in kerberos mode' do
+    let(:gssapi_mock) { double }
+    let(:user) { create(:confirmed_user) }
+
+    before do
+      @before = {
+        kerberos_service_principal: CONFIG['kerberos_service_principal'],
+        kerberos_realm:             CONFIG['kerberos_realm']
+      }
+      CONFIG['kerberos_service_principal'] = 'HTTP/obs.test.com@test_realm.com'
+      CONFIG['kerberos_realm']             = 'test_realm.com'
+    end
+
+    after do
+      CONFIG['kerberos_service_principal'] = @before[:kerberos_service_principal]
+      CONFIG['kerberos_realm'] = @before[:kerberos_realm]
+
+      # ensure users get logged out
+      User.current = nil
+    end
+
+    context 'with valid ticket' do
+      before do
+        allow(gssapi_mock).to receive(:acquire_credentials)
+        allow(gssapi_mock).to receive(:accept_context).
+          with('fake_ticket').and_return(true)
+        allow(gssapi_mock).to receive(:display_name).
+          and_return("#{user.login}@test_realm.com")
+
+        allow(GSSAPI::Simple).to receive(:new).with(
+          'obs.test.com', 'HTTP', '/etc/krb5.keytab'
+        ).and_return(gssapi_mock)
+      end
+
+      context 'and confirmed user' do
+        it 'authenticates the user' do
+          get "/source/person.xml", headers: { 'X-HTTP_AUTHORIZATION' => "Negotiate #{Base64.strict_encode64('fake_ticket')}" }
+          expect(response).to have_http_status(:success)
+          expect(User.current).to eq(user)
```

Comment:
Since this is an integration test we shouldn't be asserting on `User.current` because its too specific to the internal implementation of the authentication. We should just assert on things like the response body and status.